src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
 
-*It might be also extended to provide an interface for target independent optimizations in future*
+To avoid this kind of fusing conflicts, `Snippets` allow a plugin to mark a node as `SkippedByPlugin` so it will be ignored by the `Tokenizer`. It is a plugin responsibility to mark all nodes that (can be tokenized, but) should be ignored by `Snippets` with a `SkippedByPlugin` tag (defined in [tokenization.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/include/snippets/pass/tokenization.hpp)). This is usually implemented as a separate markup transformation that is applied immediately before the tokenization. Please, refer to the [snippets_mark_skipped.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp) pass for a detailed example.
 
-#### Register allocation
+### Optimizer
 
-Canonicalized subgraph in a snippets dialect forms a basic block or region inside a snippet (kernel). Registers are allocated globally for the whole subgraph. Since all operations for a subgraph are assumed to be vector, only vector registers are allocated for the first generation of SnippetS. Linear scan register allocation algorithm is used. Register allocator is implemented as the `ov::snippets::pass::AssignRegisters` function pass and store allocated registers for each node into `rt_info`. `rt_info` for a node holds a register for Node's output. *However, this part should be refactored better, either to become target independent or to use target-specific abstraction to acquire a new register*
+As briefly discussed in the ***Architecture*** section, `Optimizer` consists of two major units: the first one performs data flow optimization, and the second one is focused on control flow. Note however that some data-flow-related passes can be performed only after the control flow optimizations, so the second unit modifies the dataflow as well. Nevertheless, we will refer to the units as `Data flow optimizer` and `Control flow optimizer` to reflect their main purpose. Keep mind that, as discussed above, the `Data flow optimizer` operates exclusively on the `nGraph IR`, while the `Control flow optimizer` works with the `Linear IR`. We will discuss these units in more detail below.
 
-#### Schedule generation
+#### Data flow optimizer
```

Comment:
Should we add a couple of words about precision propagation?

Reply:
Agreed. Moreover, we need to create a dedicated section, where precision handling is explained in general, not only the precision propagation pass. 
A ticket for this: 111899